src/libpriv/rpmostree-util.c

```diff
@@ -434,20 +434,35 @@ rpmostree_str_has_prefix_in_ptrarray (const char *str,
   return FALSE;
 }
 
+/* Like g_strv_contains() but for ptrarray */
+gboolean
+rpmostree_str_ptrarray_contains (GPtrArray  *strs,
+                                 const char *str)
+{
+  for (guint i = 0; i < strs->len; i++)
```

Comment:
Minor perf note; should hoist the load of `strs->len` to a local constant, since the compiler can't know that `g_str_equal()` won't clobber it.  Not that we don't have perf issues 1000x greater but :smile: